presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -230,13 +230,17 @@ public static double stYMin(@SqlType(NAME) Slice input)
         return envelope.getYMin();
     }
 
+    @SqlNullable
     @Description("Returns the cardinality of the collection of interior rings of a polygon")
     @ScalarFunction("ST_NumInteriorRing")
     @SqlType(StandardTypes.BIGINT)
-    public static long stNumInteriorRings(@SqlType(NAME) Slice input)
+    public static Long stNumInteriorRings(@SqlType(NAME) Slice input)
     {
-        OGCGeometry geometry = getGeometry(deserialize(input), "ST_NumInteriorRing", POLYGON);
-        return ((OGCPolygon) geometry).numInteriorRing();
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_NumInteriorRing", POLYGON, MULTI_POLYGON);
```

Comment:
This must be a type. ST_NumInteriorRing applies only to polygons. It doesn't apply to multi-polygon. Also, the implementation casts geometry to OGCPolygon. This case will fail if geometry is multi-polygon. Right?